library/src/main/java/com/directions/route/AbstractRouting.java

```diff
@@ -111,7 +114,13 @@ private void dispatchOnCancelled() {
      */
     @Override
     protected ArrayList<Route> doInBackground(Void... voids) {
-        return new GoogleParser(constructURL()).parse();
+        ArrayList<Route> result = null;
```

Comment:
As a best pratice, create an empty list. `new GoogleParser(constructURL()).parse();` returns an empty list if it doesn't find anything.